File: backend/data/generated-driver-examples/rotate-matrix/driver.py
import sys
import json
from typing import List, Optional, Any


def main():
    lines = [line.strip() for line in sys.stdin if line.strip()]
    if not lines:
        return
    
    solution = Solution()
    idx = 0
    while idx < len(lines):
        matrix = json.loads(lines[idx + 0])
        solution.rotate(matrix)
        print("null")
        idx += 1

if __name__ == "__main__":
    main()

File: backend/data/generated-driver-examples/rotate-matrix/driver.ts
import * as fs from 'fs';

function main(): void {
  const input = fs.readFileSync(0, 'utf-8');
  const lines = input.split('\n').map(l => l.trim()).filter(l => l.length > 0);
  if (lines.length === 0) return;

  let idx = 0;
  while (idx < lines.length) {
    const matrix: number[][] = JSON.parse(lines[idx + 0]);
    rotate(matrix);
    console.log("null");
    idx += 1;
  }
}

main();